frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -520,99 +520,111 @@ function ReaderHighlight:onHoldRelease()
+                        end,
+                    },
+                    {
+                        text = _("View HTML"),
+                        enabled = not self.ui.document.info.has_pages,
+                        callback = function()
+                            self:viewSelectionHTML()
+                        end,
+                    },
                 },
                 {
-                    text = _("Search"),
-                    callback = function()
-                        self:onHighlightSearch()
-                        UIManager:close(self.highlight_dialog)
-                    end,
+                    {
+                        text = _("Wikipedia"),
+                        callback = function()
+                            UIManager:scheduleIn(0.1, function()
+                                self:lookupWikipedia()
+                                -- We don't call self:onClose(), we need the highlight
+                                -- to still be there, as we may Highlight it from the
+                                -- dict lookup widget
+                            end)
+                        end,
+                    },
+                    {
+                        text = _("Dictionary"),
+                        callback = function()
+                            self:onHighlightDictLookup()
+                            -- We don't call self:onClose(), same reason as above
+                        end,
+                    },
                 },
-            },
-        }
-        if self.selected_link ~= nil then
-            table.insert(highlight_buttons, { -- for now, a single button in an added row
                 {
-                    text = _("Follow Link"),
-                    callback = function()
-                        self.ui.link:onGotoLink(self.selected_link)
-                        self:onClose()
-                    end,
+                    {
+                        text = _("Translate"),
+                        callback = function()
+                            self:translate(self.selected_text)
+                            -- We don't call self:onClose(), so one can still see
+                            -- the highlighted text when moving the translated
+                            -- text window, and also if NetworkMgr:promptWifiOn()
+                            -- is needed, so the user can just tap again on this
+                            -- button and does not need to select the text again.
+                        end,
+                    },
+                    {
+                        text = _("Search"),
+                        callback = function()
+                            self:onHighlightSearch()
+                            UIManager:close(self.highlight_dialog)
+                        end,
+                    },
                 },
-            })
+            }
+            if self.selected_link ~= nil then
```

Comment:
Sorry, I missed this one before. You can just write:

```suggestion
            if self.selected_link then
```